google-cloud-translate/src/main/java/com/google/cloud/translate/TranslateOptions.java

```diff
@@ -135,7 +136,19 @@ public TranslateOptions build() {
 
   private TranslateOptions(Builder builder) {
     super(TranslateFactory.class, TranslateRpcFactory.class, builder, new TranslateDefaults());
-    this.apiKey = builder.apiKey != null ? builder.apiKey : getDefaultApiKey();
+    if (builder.apiKey != null) {
+      this.apiKey = builder.apiKey;
+    } else if (System.getenv(ServiceOptions.CREDENTIAL_ENV_NAME) == null) {
+      this.apiKey = getDefaultApiKey();
+    } else {
```

Comment:
This doesn't appear to be what your warning says it is.  It looks like this case is, You  haven't defined an APIKey && you've defined `GOOGLE_APPLICATION_CREDENTIALS`  -- It also doesn't appear to do anything useful.